### ISSUE TITLE
encoder output

### ISSUE THREAD

**Issue by MohamedElsaeidy** [ethz-asl/multipoint#7]
during training the encoder acts as follows and training is going fine.
<pre>input to enc =  torch.Size([1, 1, 240, 320])
output of enc =  torch.Size([1, 128, 30, 40])
at loss calculation
logits size=  torch.Size([1, 65, 30, 40])
labels size=  torch.Size([1, 30, 40])
</pre>
but when evaluating ( predicting ) the encoder output is different
<pre>input to enc =  torch.Size([1, 1, 512, 512])
output of enc =  torch.Size([1, 128, 64, 64])</pre> 

i made custom encoder. but its output shape remains same in prediction
<pre> input to enc =  torch.Size([1, 1, 512, 512])
 output of enc =  torch.Size([1, 128, 30, 40])</pre>
and the out_optical and out_thermal shapes are somehow still in that shape after passing to net()
<pre>  torch.Size([1, 1, 240, 320]) </pre>

then i got an error of different spatial dimensions at compute_descriptor_loss() because 320 doesn't match 512
right now I am training Multipoint again with input size 512 to keep fixed sizes of encoder output which is <pre> output of enc =  torch.Size([1, 128, 64, 64])</pre> 

any another suggestions to work around that issue?

**Comment by acfloria** [comment]
This is hard to tell without seeing the actual code as you seem to use a different network structure.
If you use a different encoder it should have the same output shape as the default encoder of multipoint. Otherwise you will not get the same shape as the input image when using the default decoder heads.

**Comment by MohamedElsaeidy** [comment]
i made my custom encoder output the same multipoint encoder shape which is [1, 128, 30, 40] on training, then after training while running predict_aligen_image_pair.py the mismatch happens between logits and labels

**Comment by acfloria** [comment]
The encoder just decreases the spatial dimensions by eight (achieved with the three pooling layers). So for an input with 512 pixels in each direction the encoding will have a size of [1, 128, 64, 64]. If you replace the default encoder with a custom implementation you will have to mirror these characteristics, otherwise it will not work for arbitrary image sizes (e.g. if you output a fixed size encoding)
